### PR TITLE
Added Face ID/Touch ID support and Camera support for profile pictures, fixed Official detail view social media icon size, added external link preferences

### LINKED ISSUE #17: Add FaceID/TouchID for login

### LINKED ISSUE #151: Add ability to take picture for user profile picture
Select image from library or take photo

### LINKED ISSUE #155: Setting to indicate user preference for opening links/events
Can choose whether to open all links externally, all internally, or default (links internally, events with choice)

### LINKED ISSUE #156: Improve save Settings UI
Either:
- Save setting when leaving screen (i.e. auto-save, no button)
- Kick user back to full Settings page after updating Setting

### LINKED ISSUE #157: Add back button for MapView
One that looks nice with the MapUI

### LINKED ISSUE #160: Make buttons on Officials Details VC bigger
Maybe make it like a 2x2 grid so that it utilizes more of the screen?

### PR DESCRIPTION
**Pull request by jeh97**
Closes #17 
Added ability to select profile picture with camera
Closes #151
Made officials details view social media buttons larger
Closes #160 
Added back buttons for mapView
Closes #157 
Improved settings flow
Closes #156 
Added preferences for opening external links in safari and calendar
Closes #155 

We can't use the build in iPhone keychain without signing up for and paying for the Apple Developer Program, so this is the best way I could implement Face ID/Touch ID:
When remember me is enabled, the app will keep the user logged in between sessions, if not enabled, the app will log the user out when the application is terminated.
If biometric authentication is enabled, when the app is launched and the user is still logged in due to remember me being selected, the app will attempt to use Face ID or Touch ID to authenticate the user. If the Biometric authentication is successful, the user will proceed to the Main Tab Bar View Controller, remaining logged in. If Biometric authentication is not successful, the user will be logged out and presented with the LoginViewController.

Screen Shot 2019-05-01 at 11 43 54 AM
Screen Shot 2019-05-01 at 11 44 02 AM

Screen Shot 2019-05-01 at 4 17 59 PM
Screen Shot 2019-05-01 at 4 18 02 PM
Screen Shot 2019-05-01 at 4 18 05 PM

### CODE REVIEW

File: RepresentsMe/AppDelegate.swift [hunk 1]
```diff
@@ -18,12 +18,13 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
         FirebaseApp.configure()
+        // If user still logged in
         if Auth.auth().currentUser != nil {
-            let storyboard = UIStoryboard(name: "Main", bundle: nil)
-            let viewController = storyboard.instantiateViewController(withIdentifier: "mainTabBarViewController")
-            self.window?.rootViewController = viewController
-            
-            AppState.setup()
+            // If biometric authentication enabled, require successful authentication to remain logged in
+            if Util.biometricEnabled {
```
Comment (bzsinger) [inline RepresentsMe/AppDelegate.swift]: Is this requesting the biometic authentication? Why is the if statement empty?

File: RepresentsMe/ViewControllers/AppState.swift [hunk 2]
```diff
@@ -141,6 +141,10 @@ class AppState {
     /// The listeners for sandbox address changes
     private static var sandboxAddressListeners:[OfficialsListener] = []
 
+    static var openExternalLinksInSafari:Bool = false
+    static var openCalendarOnEventExport:Bool = false
```
Comment (bzsinger) [inline RepresentsMe/ViewControllers/AppState.swift:145]: Can we make these UserDefaults/store on Firebase as a part of User?

File: RepresentsMe/ViewControllers/Entry/LoginViewController.swift [hunk 3]
```diff
@@ -42,7 +42,25 @@ class LoginViewController: UIViewController {
     /// Check if credentials were saved
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        checkRememberedCredentials()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        // If user currently logged in
+//        authenticateUserUsingBiometrics()
```
Comment (bzsinger) [inline RepresentsMe/ViewControllers/Entry/LoginViewController.swift]: Remove

File: RepresentsMe/ViewControllers/AppState.swift (hunk 2 above)
Comment (jeh97) [reply RepresentsMe/ViewControllers/AppState.swift:145]: They are stored on firebase, they are just cached here

File: RepresentsMe/ViewControllers/MapViewController.swift [hunk 4]
```diff
@@ -144,6 +157,10 @@ class MapViewController: UIViewController {
         performSegue(withIdentifier: LIST_MODAL_SEGUE_IDENTIFIER, sender: self)
     }
 
+    @objc func backButtonTapped(_ sender: Any) {
```
Comment (bzsinger) [inline RepresentsMe/ViewControllers/MapViewController.swift]: Remove?

**Review by bzsinger — APPROVED** [review]
See comments

**Comment by bzsinger** [comment]
Approved, but make sure to address comments/add commits before merging

**Pull request merged**